app/services/course/user_registration_service.rb

```diff
@@ -5,16 +5,30 @@ class Course::UserRegistrationService
   # @return [bool] True if the registration succeeded. False if the registration failed.
   def register(registration)
     CourseUser.transaction do
-      if registration.code.empty?
-        register_course_user(registration)
+      course_user = create_or_update_registration(registration)
+      succeeded = course_user && !course_user.changed?
+      if succeeded && course_user.requested?
+        notify_course_staff(registration.course, course_user)
       else
-        claim_registration_code(registration)
+        succeeded
```

Comment:
I changed notify_course_staff to return true always. That was supposed to return what deliver_now returns, but deliver_later doesn't return anything.